client/invoke.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+
+	v1 "github.com/dapr/go-sdk/dapr/proto/common/v1"
+	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/pkg/errors"
+)
+
+// InvokeService represents the request message for Service invocation
+func (c *Client) InvokeService(ctx context.Context, serviceID, method string, in []byte) (out []byte, err error) {
```

Comment:
needs to cover 4 sceanrios, please ensure all of these can be achieved here:
No Data and no return value.
No Data but a return value.
Data but no return value.
both Data and return value.

Reply:
provides additional methods for all of these combinations